src/libcrun/cgroup.c

```diff
@@ -1321,13 +1321,13 @@ write_devices_resources (int dirfd, bool cgroup2, oci_container_linux_resources_
 
   for (i = 0; i < devs_len; i++)
     {
-      cleanup_free char *fmt_buf;
+      char fmt_buf[64];
       const char *file = devs[i]->allow ? "devices.allow" : "devices.deny";
 
       if (devs[i]->type == NULL || devs[i]->type[0] == 'a')
-        len = xasprintf (&fmt_buf, "a");
+        len = sprintf (fmt_buf, "a");
       else
-        len = xasprintf (&fmt_buf, "%s %lu:%lu %s", devs[i]->type, devs[i]->major, devs[i]->minor, devs[i]->access);
+        len = snprintf (fmt_buf, sizeof (fmt_buf), "%s %lu:%lu %s", devs[i]->type, devs[i]->major, devs[i]->minor, devs[i]->access);
```

Comment:
Should this be sizeof (fmt_buf)-1?

Reply:
the `fmt_buf` is not used as a NUL terminated C string but it will be written directly to the `devices.*` file, so it won't access anything after the buffer.  Anyhow I changed it as it is clearer